### ISSUE TITLE
Dynamic import failing with TS5012 and TS6053

### ISSUE THREAD

**Issue by matt-landers** [denoland/deno#4720]
I'm hacking together a simple framework like Next.js for denobyexample.com. I'm using oak as the server framework. In one of my middleware functions, I'm loading modules dynamically like:
```
const module = await import(`../pages/${path}.tsx`)
```
This produces the following 2 errors:
```
Unhandled Error: error TS5012: Cannot read file 'file:///E:/Source/deno/denobyexample.com/pages/test.tsx': Error: assert.
►
error TS6053: File 'file:///E:/Source/deno/denobyexample.com/pages/test.tsx' not found.
```
If the import is static like:
```
const module = await import(`../pages/index.tsx`)
```
Everything works great.

Should this be supported? I saw this PR #4561 and thought it was referencing these types of scenarios.

Thanks for the help!

FYI @kitsonk

**Comment by kitsonk** [comment]
I think it is unrelated to #4561.  It is a larger defect that we should have tested for.

It relates to the situation that we statically analyse all imports, so they can be made asynchronously before we invoke the TypeScript compiler (because the compiler only supports synchronous imports).  But the preprocessing does account for all statically determinable imports, only those where the string is itself static.  So in this case the dynamic import fails.  We basically need a backup to when the preprocessing fails, as the preprocessing will always have its limits.

**Comment by bartlomieju** [comment]
Fixed in #5029
